packages/server/src/common/features/model/request-model-action-handler.ts

```diff
@@ -49,8 +50,24 @@ export class RequestModelActionHandler implements ActionHandler {
         this.logger.debug('Execute RequestModelAction:', action);
         this.modelState.setAll(action.options ?? {});
 
+        const isReconnecting = ClientOptionsUtil.isReconnecting(action.options);
+
         const progress = this.reportModelLoading('Model loading in progress');
-        await this.sourceModelStorage.loadSourceModel(action);
+
+        if (isReconnecting) {
+            const oldModelRoot = this.modelState.root;
+            if (oldModelRoot) {
+                // use current modelRoot of modelState and submit
+                this.modelState.updateRoot(oldModelRoot);
```

Comment:
Why is this necessary? We are basically resuming where we left of before the reconnect. So the modelState and indexing should already  be up to date.

Reply:
Thanks, that's true, I removed that line again!